mne/utils/check.py

```diff
@@ -771,3 +784,28 @@ def _ensure_events(events):
         raise ValueError(
             f'events must be of shape (N, 3), got {events.shape}')
     return events
+
+
+def _infer_check_export_fmt(fmt, fname, supported_formats):
+    """Infer export format from filename extension if auto.
+
+    Raises error if auto and no file extension found,
+    then checks format against supported formats, raises error if format is not
+    supported.
+    """
+    fmt = fmt.lower()
+    if fmt == "auto":
+        fmt = op.splitext(fname)[1]
+        if fmt:
+            fmt = fmt[1:].lower()
+            # find fmt in supported formats dict's tuples
+            fmt = next((k for k, v in supported_formats.items() if fmt in v),
+                       fmt)  # default to original fmt for raising error later
+        else:
+            raise ValueError(f"Couldn't infer format from filename {fname}"
+                             " (no extension found)")
+
+    if fmt not in supported_formats:
+        raise ValueError(f"Format '{fmt}' is not supported. "
+                         f"Supported formats are {supported_formats}.")
```

Comment:
I think we only want to print the keys here, separated by commas, i.e. something like

`', '.join(supported_formats.keys()`

Reply:
Well someone could attempt to use file extensions to infer automatically, and if the extension is not supported I think printing out the dictionary might be more helpful compared to just the formats.

Reply:
I can also print the dictionary in a better way (e.g. `eeglab (.set), brainvision (.eeg, .vmrk, ...)`)

Reply:
Great idea!